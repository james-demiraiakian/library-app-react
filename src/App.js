import BookList from './views/Books/BookList';
import './App.css';
import { BrowserRouter, NavLink, Switch, Route } from 'react-router-dom';
import BookDetail from './views/Books/BookDetail';

function App() {
  // TODO: Add routes to books & views
  return (
    <main className="container">
      <h1>Library Catalog</h1>
      <BrowserRouter>
        <NavLink to="/">HOME</NavLink>
        <NavLink to="/books">CATALOG</NavLink>
        <Switch>
          <Route exact path="/">
            {' '}
          </Route>
          <Route exact path="/books" component={BookList} />
          <Route exact path="/books/:id" component={BookDetail} />
        </Switch>
      </BrowserRouter>
    </main>
  );
}

export default App;
