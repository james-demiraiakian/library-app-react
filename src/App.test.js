import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

// const element = screen.getByTestId('custom-element')

it('renders a welcome message when signed out', async () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );

  screen.getByText('HOME');

  expect(container).toMatchSnapshot();
});
