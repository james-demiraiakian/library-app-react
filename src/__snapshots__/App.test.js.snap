// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a welcome message when signed out 1`] = `
<div>
  <main
    class="container"
  >
    <h1>
      Library Catalog
    </h1>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      HOME
    </a>
    <a
      href="/books"
    >
      CATALOG
    </a>
     
  </main>
</div>
`;
